src/App.tsx

```diff
@@ -1,8 +1,73 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './App.css';
+import { GoodsList } from './GoodsList/GoodsList';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const API_URL: string = 'https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json';
 
-export default App;
+const getGoods = () => {
+  return fetch(API_URL).then(res => res.json())
+}
+
+interface State {
+  listOfGoods: {id: number, name: string, color: string}[]
+}
+
+export class App extends Component<{},State> {
+  state = {
+    listOfGoods: [],
+  }
+
+  loadAllGoods = () => {
+    getGoods().then(goods => {
+      this.setState({
+        listOfGoods: goods,
+      })
+    })
+  }
+
+  loadFirstFiveGood = () => {
+    getGoods().then(goods => {
+      this.setState({
+        listOfGoods: goods.filter((good: {id: number}) => good.id <= 5),
+      })
+    })
+  }
+
+  loadRedGood = () => {
+    getGoods().then(goods => {
+      this.setState({
+        listOfGoods: goods.filter((good: {color: string}) => good.color === 'red'),
+      })
+    })
+  }
+
+  render() {
+    const { listOfGoods } = this.state;
+    return (
```

Comment:
```suggestion
    const { listOfGoods } = this.state;

    return (
```
an empty line before return/after constants block